Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm

```diff
@@ -1426,6 +1427,23 @@ - (void)_runningOrTerminatingServiceWorkerCountForTesting:(void(^)(NSUInteger))c
     });
 }
 
+NSString * const WKLocalStorage = @"WKLocalStorage";
+
+- (void)_fetchDataOfTypes:(NSSet<NSString *> *)dataTypes completionHandler:(WK_SWIFT_UI_ACTOR void (^)(NSData *))completionHandler
+{
+    if ([dataTypes containsObject:WKLocalStorage]) {
+        _websiteDataStore->fetchLocalStorage([completionHandler = makeBlockPtr(completionHandler)](HashMap<WebCore::ClientOrigin, HashMap<String, String>> localStorage) {
+            WTF::Persistence::Encoder encoder;
+            encoder << localStorage;
+            completionHandler([[NSData alloc] initWithBytes:encoder.buffer() length:encoder.bufferSize()]);
```

Comment:
The better way to do this in to include `<wtf/cocoa/SpanCocoa.h>` and use `toNSData(encoder.span())`